docs/jargon-mesh.md

```diff
@@ -0,0 +1,658 @@
+---
+id: terminology
+title: Temporal's core terminology
+sidebar_label: Terminology
+---
+
+import Mermaid from '@theme/Mermaid';
+
+## Workflow
+
+A fault-oblivious stateful function that orchestrates Activities.
+
+- A Workflow has full control over which [**Activities**](#activity) are executed, and in which order.
+- A Workflow must not affect the external world directly, only through [**Activities**](#activity).
+- What makes Workflow code a Workflow is that its state is preserved by Temporal.
+  Therefore any failure of a [**Worker**](#worker) process that hosts the Workflow code does not affect the [**Workflow Execution**](#workflow-execution).
+  The Workflow continues as if these failures did not happen.
+  At the same time, [**Activities**](#activity) can fail any moment for any reason.
+- Because Workflow code is fully fault-oblivious, it is guaranteed to get notifications about [**Activity**](#activity) failures or timeouts and act accordingly.
+- There is no limit to the duration of a Workflow Execution.
```

Comment:
I would maybe say something like "By default, workflow execution is umlimited, however you can set the workflow execution timeout to limit it" .. so user understands its not always "no limit"

Reply:
Also, even if it set to "unlimited" it can fail depending on some issues like event log size (nice way to link from here to the continueasnew section ;))